src/xwayland.c

```diff
@@ -30,6 +30,16 @@ xwayland_view_get_size_hints(struct view *view)
 	};
 }
 
+static bool
+xwayland_view_wants_focus(struct view *view)
+{
+	xcb_icccm_wm_hints_t *hints = xwayland_surface_from_view(view)->hints;
+	if (!hints) {
+		return true;
+	}
+	return (bool)hints->input;
```

Comment:
How confident are we on this logic (`hints->input`)?
Having a quick read here (https://tronche.com/gui/x/icccm/sec-4.html) makes my head spin.

Reply:
Speaking for myself, no more or less confident than anything else X11... it's all trial-and-error.
This is how `xprop` describes that flag, for what it's worth:
```
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: False
```

Reply:
There is some more complexity with a WM_TAKE_FOCUS atom that I don't fully understand either; however, I think we are using the "input" flag (which is just a Boolean, though 32 bits) correctly so far as it goes:

> Passive and Locally Active clients set the input field of WM_HINTS to True, which indicates that they require window manager assistance in acquiring the input focus. No Input and Globally Active clients set the input field to False, which requests that the window manager not set the input focus to their top-level window.

Reply:
Happy to try it :smile:
Most of the code I've written for labwc has been trial-and-error tbh. Certain things are clear in protocols, etc but there is a lot of things relating to window management that aren't exactly well documented nor obvious.

Reply:
> There is some more complexity with a WM_TAKE_FOCUS atom that I don't fully understand either; however, I think we are using the "input" flag (which is just a Boolean, though 32 bits) correctly so far as it goes:
> 
> > Passive and Locally Active clients set the input field of WM_HINTS to True, which indicates that they require window manager assistance in acquiring the input focus. No Input and Globally Active clients set the input field to False, which requests that the window manager not set the input focus to their top-level window.

Maybe worth adding the above (ish) as a comment in the code ?? to help us/others to maintain.

Reply:
Sure, I can do that.